### PR TITLE
Switch Pilot routing e2e tests to v1alpha2

### PR DESCRIPTION
**Pull request by ijsnellf**
Replaces v1alpha1 configs with v1alpha2 for the routing portion of the pilot e2e tests.

### CODE REVIEW

File: pilot/test/integration/ingress.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func (t *ingress) setup() error {
 	}
 
 	// send route rules for "c" only
-	return t.applyConfig("rule-default-route.yaml.tmpl", nil)
+	return t.applyConfig("rule-default-route-v1alpha1.yaml.tmpl", nil)
```
Comment (rshriram) [inline pilot/test/integration/ingress.go]: why v1alpha1 here?

File: pilot/test/integration/routing.go [hunk 2]
```diff
@@ -52,7 +52,7 @@ func (t *routing) run() error {
 			description: "routing all traffic to c-v1",
 			config:      "rule-default-route.yaml.tmpl",
 			check: func() error {
-				return t.verifyRouting("http", "a", "c", "", "", 100, map[string]int{"v1": 100, "v2": 0}, "default-route")
+				return t.verifyRouting("http", "a", "c", "", "", 100, map[string]int{"v1": 100, "v2": 0}, "route-rule-c")
```
Comment (rshriram) [inline pilot/test/integration/routing.go]: also, call it route-rule-a-to-cv1

Comment (rshriram) [reply pilot/test/integration/routing.go]: But then, why even change, since the name reflects the default route rule being applied

Comment (ijsnellf) [reply pilot/test/integration/routing.go]: Right now I am continually updating the v1alpha2 route rule named "route-rule-c". If we wanted to use different names, we would have to create and delete the rules.

File: pilot/test/integration/ingress.go (hunk 1 above)
Comment (ijsnellf) [reply pilot/test/integration/ingress.go]: v1alpha2 doesn't work with ingress.

File: pilot/test/integration/testdata/destination-c.yaml.tmpl [hunk 3]
```diff
@@ -0,0 +1,13 @@
+apiVersion: config.istio.io/v1alpha2
```
Comment (rshriram) [inline pilot/test/integration/testdata/destination-c.yaml.tmpl]: call it destination-rule-c.yaml.tmpl

File: pilot/test/integration/testdata/rule-default-route-mirrored.yaml.tmpl [hunk 4]
```diff
@@ -1,14 +1,15 @@
...
+  hosts:
+    - c
+  http:
+    - route:
+      - destination:
+          name: c
+          subset: v1
+        weight: 100
+      mirror:
+        name: b
```
Comment (rshriram) [inline pilot/test/integration/testdata/rule-default-route-mirrored.yaml.tmpl]: This relates to @christian-posta 's bug (https://github.com/istio/istio/issues/3270).. We shouldn't require mirror

File: pilot/test/integration/testdata/rule-fault-injection.yaml.tmpl [hunk 5]
```diff
@@ -1,27 +1,26 @@
 apiVersion: config.istio.io/v1alpha2
-kind: RouteRule
+kind: V1alpha2RouteRule
 metadata:
-  name: fault-route
```
Comment (rshriram) [inline pilot/test/integration/testdata/rule-fault-injection.yaml.tmpl]: why remove the names? They are supposed to be useful for the decorators and help people understand. Unless you want each route to override other.

Comment (rshriram) [reply pilot/test/integration/testdata/rule-fault-injection.yaml.tmpl]: Oh, never mind... ignore the comment.

File: pilot/test/integration/testdata/rule-default-route-mirrored.yaml.tmpl (hunk 4 above)
Comment (ijsnellf) [reply pilot/test/integration/testdata/rule-default-route-mirrored.yaml.tmpl]: Agreed. This is due to a bug in the implementation of shadow cluster that was copied over from v1alpha1. Should be an easy fix.

**Comment by costinm** [comment]
I don't think we should replace - we need to test both.

**Comment by rshriram** [comment]
We can't, unless we add in more kludge. Also, its v1alpha1/v1alpha2 :). People are getting confused about two forms of route rules, etc. We are going to write a tool to convert from old to new. So that should help

**Pull request closed without merging**